### ISSUE TITLE
Input issue again...

### ISSUE THREAD

**Issue by NonstickAtom785** [Zeda/Grammer2#37]
I don't have the slightest clue why but Input is putting a bunch of random characters right after the cursor. I thought I had mistyped something but the syntax seems right.

It happens on my font program. Tried making a search utility and discovered it.

**Comment by Zeda** [comment]
I'm not replicating this bug, do you mind posting the code? I'm wondering if you are trying to do something like `Input A`, if you mean to do `Input →A` since `Input ` takes an optional argument for a trailing string. For example:
```
")→S
Text(0,0,"(
Input S→A
```
This would draw a `)` after whatever you are inputting, essentially making it appear to the user to be wrapped in `()`.

**Comment by NonstickAtom785** [comment]
Um okay. So running 
```
expr(Input "→F
```
Does something different as apposed to
```
expr(Input "
→F
```
That one does what I want it to do.
But my previous code where I put just an `Input` without the quote...
```
expr(Input →F
```
Does the weird stuff.
It reminds me of when you told me that the expr( is parsing the F too.

**Comment by Zeda** [comment]
I'm using the above codes and I am still not replicating the bug you are reporting.
There are two different bugs that your code revealed though, actually, just quite the opposite of what you are finding-- it's drawing *nothing* after the cursor when it should be drawing *something*. 

Also, by using an empty string like that, apparently the parser thinks it should include at least one byte. It should not do this. A quick remedy us just `""` which is annoying,so sorry about that. Also, why ate you using an empty string to close the text? that's default behavior.

To summarize:
- I am not seeing the behavior of printing random characters after the cursor.
- In fact, I am seeing it not print any characters, even when it is supposed to.
- `"→` is not behaving as an empty string the way it should.

So which version of Grammer do you have? It's weird that you are seeing random chars.

**Comment by NonstickAtom785** [comment]
The latest Grammer is being used. I was using the empty string at first that's where the issue is lying.
I'm getting the extra chars at the end when I'm giving input for GFNTEDT and OMNIFNT. I fixed it by adding a `"` and moving the `→` down to the next line.

**Comment by Zeda** [comment]
Okay, that helps. I am able to replicate it; these programs relocate the `Input `. Now I'll be able to fix it, thanks.

**Comment by Zeda** [comment]
Hopefully these have fixed it.

**Comment by NonstickAtom785** [comment]
Yah I was going to give you my new thoughts about what was causing it. Which was setting the input buffer did something to cause the behavior. But you figured it out! Thanks!
